Website/DesktopModules/Admin/Security/Register.ascx.cs

```diff
@@ -103,270 +104,72 @@ protected bool IsValid
 			}
 		}
 
-		protected string UserToken
+		protected string RedirectURL
 		{
 			get
 			{
-				return ViewState.GetValue("UserToken", string.Empty);
-			}
-			set
-			{
-				ViewState.SetValue("UserToken", value, string.Empty);
-			}
-		}
+				string redirectUrl = "";
 
-		#endregion
-
-		#region Event Handlers
-
-		protected override void OnInit(EventArgs e)
-		{
-			base.OnInit(e);
-
-			jQuery.RequestDnnPluginsRegistration();
-
-			ClientResourceManager.RegisterScript(Page, "~/Resources/Shared/scripts/dnn.jquery.extensions.js");
-			ClientResourceManager.RegisterScript(Page, "~/Resources/Shared/scripts/dnn.jquery.tooltip.js");
-			ClientResourceManager.RegisterScript(Page, "~/DesktopModules/Admin/Security/Scripts/dnn.PasswordComparer.js");
-
-			if (PortalSettings.Registration.RegistrationFormType == 0)
-			{
-				//UserName
-				if (!PortalSettings.Registration.UseEmailAsUserName)
+				if (PortalSettings.Registration.RedirectAfterRegistration > 0) //redirect to after registration page
 				{
-
-					AddField("Username", String.Empty, true,
-							String.IsNullOrEmpty(PortalSettings.Registration.UserNameValidator) ? ExcludeTerms : PortalSettings.Registration.UserNameValidator,
-							TextBoxMode.SingleLine);
-				}
-
-				//Password
-				if (!PortalSettings.Registration.RandomPassword)
-				{
-					AddPasswordStrengthField("Password", "Membership", true);
-
-					if (PortalSettings.Registration.RequirePasswordConfirm)
-					{
-						AddPasswordConfirmField("PasswordConfirm", "Membership", true);
-					}
-				}
-
-				//Password Q&A
-				if (MembershipProviderConfig.RequiresQuestionAndAnswer)
-				{
-					AddField("PasswordQuestion", "Membership", true, String.Empty, TextBoxMode.SingleLine);
-					AddField("PasswordAnswer", "Membership", true, String.Empty, TextBoxMode.SingleLine);
-				}
-
-				//DisplayName
-				if (String.IsNullOrEmpty(PortalSettings.Registration.DisplayNameFormat))
-				{
-					AddField("DisplayName", String.Empty, true, String.Empty, TextBoxMode.SingleLine);
+					redirectUrl = Globals.NavigateURL(PortalSettings.Registration.RedirectAfterRegistration);
 				}
 				else
 				{
-					AddField("FirstName", String.Empty, true, String.Empty, TextBoxMode.SingleLine);
-					AddField("LastName", String.Empty, true, String.Empty, TextBoxMode.SingleLine);
-				}
 
-				//Email
-				AddField("Email", String.Empty, true, PortalSettings.Registration.EmailValidator, TextBoxMode.SingleLine);
-
-				if (PortalSettings.Registration.RequireValidProfile)
-				{
-					foreach (ProfilePropertyDefinition property in User.Profile.ProfileProperties)
+					if (PortalSettings.Registration.RedirectAfterRegistration <= 0)
```

Comment:
this is inside of the `else` for `if (PortalSettings.Registration.RedirectAfterRegistration > 0)`, so it will always be `true`.  is there some particular case you're trying to address by adding this `if` check?